controller/api/destination/server_test.go

```diff
@@ -158,6 +159,110 @@ func TestGet(t *testing.T) {
 		}
 	})
 
+	t.Run("Return endpoint opaque protocol controlled by a server", func(t *testing.T) {
+		server, client := getServerWithClient(t)
+		defer server.clusterStore.UnregisterGauges()
+
+		stream := &bufferingGetStream{
+			updates:          make(chan *pb.Update, 50),
+			MockServerStream: util.NewMockServerStream(),
+		}
+		defer stream.Cancel()
+		errs := make(chan error)
+
+		path := fmt.Sprintf("%s:%d", fullyQualifiedNamePolicy, 80)
+
+		// server.Get blocks until the grpc stream is complete so we call it
+		// in a goroutine and watch stream.updates for updates.
+		go func() {
+			err := server.Get(&pb.GetDestination{
+				Scheme: "k8s",
+				Path:   path,
+			}, stream)
+			if err != nil {
+				errs <- err
+			}
+		}()
+
+		select {
+		case err := <-errs:
+			t.Fatalf("Got error: %s", err)
+		case update := <-stream.updates:
+			addrs := update.GetAdd().Addrs
+			if len(addrs) == 0 {
+				t.Fatalf("Expected len(addrs) to be > 0")
+			}
+
+			if addrs[0].GetProtocolHint().GetOpaqueTransport() == nil {
+				t.Fatalf("Expected opaque transport for %s but was nil", path)
+			}
+		}
+
+		// Update the Server's pod selector so that it no longer selects the
+		// pod. This should result in the proxy protocol no longer being marked
+		// as opaque.
+		srv := v1beta2.Server{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "srv",
+				Namespace: "ns",
+			},
+			Spec: v1beta2.ServerSpec{
+				PodSelector: &metav1.LabelSelector{
+					// PodSelector is updated to NOT select the pod
+					MatchLabels: map[string]string{"app": "FOOBAR"},
+				},
+				ExternalWorkloadSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{"app": "external-workload-policy-test"},
+				},
```

Comment:
what is the significance of having the ExternalWorkloadSelector selector set here in this test. Isn't there a one-of constraint present on the podSelector and externalWorkloadSelector fields in general